core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -90,11 +92,13 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
    * @param config The config of the topic
    * @param partitionReplicaAssignment The assignments of the topic
    * @param validate Boolean indicating if parameters must be validated or not (true by default)
+   * @param usesTopicId Boolean indicating whether the topic ID will be created
    */
   def createTopicWithAssignment(topic: String,
                                 config: Properties,
                                 partitionReplicaAssignment: Map[Int, Seq[Int]],
-                                validate: Boolean = true): Unit = {
+                                validate: Boolean = true,
+                                usesTopicId: Boolean = false): Unit = {
```

Comment:
So if we use `TopicCommand` with `ZookeeperTopicService` to create a topic, we will always not set a topicId?

Reply:
Initially yes. The controller will process and add topic IDs to these topics if IBP is at least 2.8